### PR TITLE
Allow customization of shim output root

### PR DESCRIPTION
**Pull request by wli3**
So you could customize the output like the following
``` xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <PackAsToolShimRuntimeIdentifiers>win-x64;osx.10.12-x64</PackAsToolShimRuntimeIdentifiers>
    <PackAsTool>true</PackAsTool>
    <PackagedShimOutputRootDirectory>bin/$(Configuration)</PackagedShimOutputRootDirectory>
  </PropertyGroup>

</Project>
```

note you cannot use <PackagedShimOutputRootDirectory>$(OutDir)</PackagedShimOutputRootDirectory> since OutDir is not set by the time of reading project file

### CODE REVIEW

**Comment by wli3** [comment]
@mikem8361 can this unblock you?

**Comment by mikem8361** [comment]
Yes. 

/cc: @jaredpar

**Review by nguerrera — APPROVED** [review]

**Comment by wli3** [comment]
@dsplaisted need your approval for 2.1.4xx

**Comment by johnbeisner** [comment]
@dotnet-bot test Windows_NT x64 sdk perf tests

**Comment by wli3** [comment]
@Pilchie could you approve this to go 2.1.4xx

**Comment by wli3** [comment]
ping @Pilchie

**Comment by jaredpar** [comment]
Why would this not be the default value? Won't every single customer that does signing need to do this?

**Comment by wli3** [comment]
@jaredpar it would be this PR https://github.com/dotnet/sdk/pull/2366 . (WIP is used to mark holding for preview 5, so I closed it instead of title it WIP)

here is why the proper fix ends up complicated

1. pack produce file in `bin` is unexpected, only build should do it
2. and if we want to generate shims in publish we need to support publish --no-build, it is not possible to do it in publish. So we need to put generate shims in `build`
3. but put stuff in build has much higher bar
3.1 need to float asset to publish and pack
3.2. need to support fast incremental build, used cached asset file data.
3.3 need to support `dotnet clean`

so we push up a fix that can unblock first

**Comment by jaredpar** [comment]
@wli3 is the eventual goal to make this on by default?

**Comment by nguerrera** [comment]
@jaredpar Yes, this will effectively be on by default when we do the real fix and generate the shims in build.

**Comment by jaredpar** [comment]
@nguerrera great! What issue is tracking that?

**Comment by nguerrera** [comment]
@jaredpar Still https://github.com/dotnet/cli/issues/9288 This won't close that.

**Comment by wli3** [comment]
@Pilchie this change does not meet the bar for 15.8?

cc @mikem8361 @jaredpar @danroth27

**Comment by Pilchie** [comment]
Sorry - not sure how I missed this.  Approved, as long as we can get the insertion done by 6pm today...

**Comment by mikem8361** [comment]
What version of the sdk will this be in?

**Comment by wli3** [comment]
@mikem8361 if we managed to insert it into VS it will be in 2.1.400. Preview build is still in CI. No, version number yet

**Comment by mikem8361** [comment]
Can you let me know the version when CI is done and the package is published. Thanks.

**Comment by wli3** [comment]
@mikem8361 I am sorry for the delay. The latest should have it so you can get it from this page https://github.com/dotnet/cli/tree/release/2.1.4xx  (any 2.14xx version above 2.1.400-preview-009085)

**Pull request merged**
